DeDriftAndResample/DeDriftAndResamplePipeline.sh

```diff
@@ -408,75 +410,46 @@ for Mesh in ${LowResMeshes} ${HighResMesh} ; do
 
 	# Resample scalar maps and apply new bias field
 	log_Msg "Resample scalar maps and apply new bias field"
-
-	for Map in ${Maps} ${MyelinMaps} SphericalDistortion ArealDistortion EdgeDistortion StrainJ StrainR ; do
+	
+	BiasFieldComputed=false
+	MyelinMapsToUse=""
+	# myelin map only loop
+	for MyelinMap in ${MyelinMaps} ; do
+		if [ "$BiasFieldComputed" = false ]; then
+			# ----- Begin moved statements -----
+			# Recompute Myelin Map Bias Field Based on Better Registration
+			log_Msg "Recompute Myelin Map Bias Field Based on Better Registration"
+			log_Debug_Msg "Point 1.1"
+			
+			# Myelin Map BC using low res
+			"$HCPPIPEDIR"/global/scripts/MyelinMap_BC.sh \
+				--study-folder="$StudyFolder" \
+				--subject="$Subject" \
+				--registration-name="$OutputRegName" \
+				--msm-all-templates="$MSMAllTemplates" \
+				--use-ind-mean="$UseIndMean" \
+				--low-res-mesh="$LowResMesh" \
+				--myelin-target-file="$MyelinTargetFile" \
+				--map="$MyelinMap"
+			# ----- End moved statements -----
+			# bias field is computed in the module MyelinMap_BC.sh
+			BiasFieldComputed=true
+		else
+			# bias field in native space is already generated
+			# BC the other types of given myelin maps
+			${Caret7_Command} -cifti-math "Var - Bias" ${NativeFolder}/${Subject}.${MyelinMap}_BC_${OutputRegName}.native.dscalar.nii -var Var ${NativeFolder}/${Subject}.${MyelinMap}.native.dscalar.nii -var Bias ${NativeFolder}/${Subject}.BiasField_${OutputRegName}.native.dscalar.nii
+		fi
+		MyelinMapsToUse+="${MyelinMap}_BC "
+	done
+	
+	log_Debug_Msg "Point 2.0"
+	if [ "$BiasFieldComputed" = false ]; then
```

Comment:
This can only happen if `MyelinMaps` is empty, which already matches the initialization of `MyelinMapsToUse`, so this isn't needed.